tests/EdgeToEdge/Routes/ApplyForMembershipRouteTest.php

```diff
@@ -178,4 +180,50 @@ public function testGivenValidRequest_requestIsRedirected() {
 		} );
 	}
 
+	public function testWhenApplicationGetsPersisted_timestampIsStoredInCookie() {
+		$this->createEnvironment( [], function ( Client $client, FunFunFactory $factory ) {
+			$factory->setNullMessenger();
+			$factory->setCookieHandler( new FixedCookieHandler( [ 'memapp_timestamp' => self::FIXED_TIMESTAMP ] ) );
+
+			$client->request(
+				'POST',
+				'/apply-for-membership',
+				$this->newValidHttpParameters()
+			);
+
+			$this->assertSame(
+				self::FIXED_TIMESTAMP,
+				$this->getCookieByName( $client->getResponse()->headers->getCookies(), 'memapp_timestamp' )
+			);
+		} );
+	}
+
+	private function getCookieByName( array $cookies, string $name ) {
+		foreach ( $cookies as $cookie ) {
+			if ( $cookie->getName() === $name ) {
+				return $cookie->getValue();
+			}
+		}
+
+		return false;
+	}
+
+	public function testWhenMultipleMembershipFormSubmissions_requestGetsRejected() {
+		$this->createEnvironment( [], function ( Client $client, FunFunFactory $factory ) {
+			$factory->setNullMessenger();
+
+			$cookieHandler = new FixedCookieHandler( [ 'memapp_timestamp' => self::FIXED_TIMESTAMP ] );
```

Comment:
Would't it be better to use a timestamp 10 seconds ago? Otherwise the tests will break in 2020.